### ISSUE TITLE
Unused variable warning for default trait methods

### ISSUE THREAD

**Issue by CryZe** [rust-lang/rust#92696]
Given the following code: [Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=a30cffae65480766a5dc1336a5b1a7cf)

```rust
pub trait AutoSplitter {
    type Data;
    fn update(&mut self) -> Option<Self::Data>;
    fn should_start(data: &Self::Data) -> bool {
        false
    }
}
```

The current output is:

```
warning: unused variable: `data`
 --> src/lib.rs:4:21
  |
4 |     fn should_start(data: &Self::Data) -> bool {
  |                     ^^^^ help: if this is intentional, prefix it with an underscore: `_data`
  |
  = note: `#[warn(unused_variables)]` on by default
```

Ideally the compiler should not emit any warnings as prefixing the parameter with an underscore would negatively affect the documentation and auto generated implementations via an IDE / rust-analyzer.

**Comment by ehuss** [comment]
I believe this is a duplicate of #64723, #26487, and #14027.  Per https://github.com/rust-lang/rust/issues/26487#issuecomment-169828205, it sounds like the lang team would prefer to not make this change.

**Comment by ehuss** [comment]
Closing as a duplicate of the above linked issues.
